src/main/java/org/spongepowered/common/world/WorldManager.java

```diff
@@ -1205,30 +1222,31 @@ public static void sendDimensionRegistration(EntityPlayerMP playerMP, WorldProvi
     }
 
     public static void loadDimensionDataMap(@Nullable NBTTagCompound compound) {
-        dimensionBits.clear();
+        usedDimensionIds.clear();
+        lastUsedDimensionId = 0;
+
         if (compound == null) {
-            dimensionTypeByDimensionId.keySet().stream().filter(dimensionId -> dimensionId >= 0).forEach(dimensionBits::set);
+            dimensionTypeByDimensionId.keySet().stream().filter(dimensionId -> dimensionId >= 0).forEach(usedDimensionIds::add);
         } else {
-            final int[] intArray = compound.getIntArray("DimensionArray");
+            for (int id : compound.getIntArray("UsedIDs")) {
```

Comment:
If you can store these constants under `NbtDataUtil.Forge`, that'd be perfect (it's where we store a bunch of string constants for NbtData, so we have clearer migrations if we can provide ways to upgrade data from forge to sponge vanilla and vice versa).